client/modules/accounts/components/passwordReset/updatePasswordOverlay.js

```diff
@@ -19,14 +19,23 @@ class UpdatePasswordOverlay extends Component {
     super();
 
     this.state = {
-      password: ""
+      password: "",
+      showSpinner: true
     };
 
     this.handleFieldChange = this.handleFieldChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
     this.handleCancel = this.handleCancel.bind(this);
   }
 
+  componentDidMount() {
```

Comment:
Using `setTimeout` is considered bad practice and you really shouldn't be using it anywhere in the project, especially on UI programming.